core/trino-main/src/main/java/io/trino/sql/planner/optimizations/BeginTableWrite.java

```diff
@@ -72,7 +74,19 @@ public BeginTableWrite(Metadata metadata)
     @Override
     public PlanNode optimize(PlanNode plan, Session session, TypeProvider types, SymbolAllocator symbolAllocator, PlanNodeIdAllocator idAllocator, WarningCollector warningCollector)
     {
-        return SimplePlanRewriter.rewriteWith(new Rewriter(session), plan, Optional.empty());
+        try {
+            return SimplePlanRewriter.rewriteWith(new Rewriter(session), plan, Optional.empty());
+        }
+        catch (RuntimeException e) {
+            try {
+                int nestLevel = 4; // so that it renders reasonably within exception stacktrace
+                String explain = textLogicalPlan(plan, types, metadata, StatsAndCosts.empty(), session, nestLevel, false);
+                e.addSuppressed(new Exception("Current plan:\n" + explain));
+            }
+            catch (RuntimeException ignore) {
+            }
```

Comment:
nit: connect with previous line

Reply:
@losipiuk https://github.com/trinodb/trino/pull/8145 :)

Reply:
oops :)
Weird indeed that we allow that for method bodies and not here :)